src/components/Typography/Typography.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+import styles from './Typography.module.css';
+
+type TagType =
+  | keyof JSX.IntrinsicElements
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | React.FC<any>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | React.ComponentClass<any>;
+
+type Color = 'primary' | 'primary-alt' | 'black';
+type Weight = 'normal' | 'bold';
+type Align = 'inherit' | 'left' | 'center' | 'right' | 'justify';
+type Size = 'sm' | 'md' | 'lg' | 'xl' | 'xl2' | 'xl3';
+
+interface TypographyProps {
+  align?: Align;
+  children?: React.ReactNode;
+  color?: Color;
+  className?: string;
+  tag?: TagType;
+  weight?: Weight;
+  size?: Size;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
+}
+
+type TypographyType = React.FC<TypographyProps> & {
+  Block: React.FC<TypographyProps>;
+};
+const Typography: TypographyType = ({
+  align = 'inherit',
+  color = 'primary',
+  className: passedClassName = '',
+  size = 'md',
+  tag: Tag = 'span',
+  weight = 'normal',
+  ...props
+}) => {
+  const className = useMemo(
+    () =>
+      classNames(
+        styles[color],
+        styles[weight],
+        styles[align],
+        styles[size],
+        passedClassName
+      ),
+    [align, color, passedClassName, weight, size]
+  );
+
+  return <Tag {...props} className={className} />;
```

Comment:
This is interesting! Does this work because html element names are actually calls to `createElement` in jsx? 

What happen if someone passes in an invalid element name or one that isn't supported?

Reply:
I _think_ it works that way, but I didn't read the source.

I first stole if from [reactstrap](https://github.com/reactstrap/reactstrap/blob/master/types/lib/Button.d.ts) (React Bootstrap). I love it for things like buttons that can be links, or typography that can be just about anything. imo, nice to read, super flexible.

It can be typed with [React.ElementType](https://flow.org/en/docs/react/types/#toc-react-elementtype)